### ISSUE TITLE
OpenJDK 10/11

### ISSUE THREAD

**Issue by hjacobs** [zalando/docker-openjdk#20]
OpenJDK 10/11 is part of Ubuntu 18.04 LTS ("bionic"): https://packages.ubuntu.com/bionic/openjdk-11-jdk

* http://openjdk.java.net/projects/jdk/10/
* http://openjdk.java.net/projects/jdk/11/

See also the update to 18.04: https://github.com/zalando-stups/docker-ubuntu/pull/22

I propose making OpenJDK 10/11 the default for this Docker base image (without changing its name).

**Comment by aermakov-zalando** [comment]
It might break a lot of applications that still rely on private APIs (like `sun.misc.Unsafe`). I would suggest to publish a separate image for testing first, because the migration path might be non-trivial (or impossible if applications rely on unsupported libraries).

**Comment by hjacobs** [comment]
@aermakov-zalando yes, I think for migration purposes it makes sense to gradually release with a new name, but mid-term I would change the default (people can always pin their version to a known good Docker image tag).

**Comment by kgalli** [comment]
@hjacobs will Java 11 be soon available? Any further plan if it will be the default or like suggested a separate image. Would like to start migration process as soon as possible. Can even create the necessary PR just need to know which direction is preferred.

**Comment by Jan-M** [comment]
:+1: for another image name with the new major version of Java.

**Comment by harti2006** [comment]
Since Java 10 there, the JVM is aware of running in (Docker) containers. This is extremely useful for applications running on Kubernetes to not get OOMKilled every couple of minutes.

**Comment by AlexanderYastrebov** [comment]
[JDK-8186248 : Allow more flexibility in selecting Heap % of available RAM](https://bugs.java.com/view_bug.do?bug_id=JDK-8186248)

**Comment by lalib** [comment]
@hjacobs @harti2006 Is there any plan or roadmap to support OpenJDK 11 ?

**Comment by hjacobs** [comment]
@lalib we don't actively work on it, if anybody wants to pick this up, feel free :smile:

**Comment by irishshagua** [comment]
@hjacobs 
I've opened a PR to add a [JDK 11 build](https://github.com/zalando-stups/docker-openjdk/pull/24) if you get a chance to review.
I was hoping it would auto-trigger a build. Not sure if there is some other process for that 😕

**Comment by vadeg** [comment]
Hello,

Any updates on this?

**Comment by hjacobs** [comment]
@vadeg please look into our internal Zalando Developer Console and search for "jdk11", you will find a migration guide (which does not rely on this base Docker image anymore).

**Comment by vadeg** [comment]
@hjacobs thanks
